river/command/map.zig

```diff
@@ -211,3 +218,76 @@ fn parseOptionalArgs(args: []const []const u8) OptionalArgsContainer {
 
     return parsed;
 }
+
+/// Remove a mapping from a given mode
+///
+/// Example:
+/// unmap normal Mod4+Shift Return
+pub fn unmap(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    const optionals = parseOptionalArgs(args[1..]);
+    // offset caused by optional arguments
+    const offset = optionals.i;
+    if (args.len - offset < 4) return Error.NotEnoughArguments;
+
+    const mode_id = try modeNameToId(allocator, seat, args[1 + offset], out);
+    const modifiers = try parseModifiers(allocator, args[2 + offset], out);
+    const keysym = try parseKeysym(allocator, args[3 + offset], out);
+
+    // Find the already existing mapping if it exists
+    const mode_mappings = &seat.input_manager.server.config.modes.items[mode_id].mappings;
+    const mapping_idx = for (mode_mappings.items) |existant_mapping, i| {
+        if (existant_mapping.modifiers == modifiers and existant_mapping.keysym == keysym and existant_mapping.release == optionals.release) {
+            break i;
+        }
+    } else {
+        out.* = try std.fmt.allocPrint(
+            allocator,
+            "there is no mapping for modifiers '{}' and keysym '{}'",
+            .{ args[2 + offset], args[3 + offset] },
+        );
+        return Error.Other;
+    };
```

Comment:
Could we de-duplicate this check across `map`/`unmap` and `map-pointer`/`unmap-pointer`?

Reply:
Done.
It didn't deduplicate that much because the error messages are different.